src/main/kotlin/tanvd/grazi/spellcheck/GraziSpellchecker.kt

```diff
@@ -96,4 +102,30 @@ object GraziSpellchecker : GraziLifecycle {
     override fun reset() = init()
 
     override fun reInit() = init()
+
+    override fun update(prevState: GraziConfig.State?, newState: GraziConfig.State, project: Project) {
+        if (prevState?.enabledSpellcheck == newState.enabledSpellcheck) return
+
+        if (newState.enabledSpellcheck) {
+            if (!ApplicationManager.getApplication().isUnitTestMode) {
+                modifyAndCommitProjectProfile(project, Consumer {
+                    it.getTools(SpellCheckingInspection.SPELL_CHECKING_INSPECTION_TOOL_NAME, project).isEnabled = false
+                })
+            }
+            with(CommitMessageInspectionProfile.getInstance(project)) {
+                getTools(getTool(CommitMessageSpellCheckingInspection::class.java).shortName, project).isEnabled = false
+            }
+        } else {
+            with(CommitMessageInspectionProfile.getInstance(project)) {
+                getTools(getTool(CommitMessageSpellCheckingInspection::class.java).shortName, project).isEnabled = true
+            }
+        }
+
+        //enable back built-in spellcheck if it is not initialization, previous state was `enabled` and new is `disabled`
+        if ((prevState?.enabledSpellcheck == true && !newState.enabledSpellcheck) && !ApplicationManager.getApplication().isUnitTestMode) {
```

Comment:
Strange condition, if newState. enabledSpellcheck is false, prevState?.enabledSpellcheck is 100% true, because of 107 line.

Reply:
Also, you can combine it with upper one